src/test/perl/modules/filetools.pm

```diff
@@ -0,0 +1,31 @@
+package filetools;
+
+use strict;
+use warnings;
+
+use parent qw(Exporter);
+use File::Basename qw(dirname);
+use File::Path qw(mkpath);
+
+our @EXPORT_OK = qw(readfile makefile);
+
+# cannot use mocked filewriter
+sub makefile
+{
+    my $fn = shift;
+    my $dir = dirname($fn);
+    mkpath $dir if ! -d $dir;
+    open(FH, ">$fn");
```

Comment:
Bare filehandles and two element open is considered dangerous. Please use ```open(my $fh, ">", "$fn");``` as per the Perl Best Practice Book.